crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -465,22 +468,20 @@ impl CumulocityConverter {
             .from_smartrest(smartrest)?
             .to_thin_edge_json()?;
 
-        // Pass the fresh token to the tedge-agent as it cannot request a new one
-        let token = self.http_proxy.get_fresh_jwt_token().await?;
-
         software_update_request
             .update_list
             .iter_mut()
             .for_each(|modules| {
                 modules.modules.iter_mut().for_each(|module| {
-                    if let Some(url) = &module.url {
-                        if self.c8y_endpoint.url_is_in_my_tenant_domain(url.url()) {
-                            module.url = module.url.as_ref().map(|s| {
-                                DownloadInfo::new(&s.url).with_auth(Auth::new_bearer(&token))
-                            });
+                    if let Some(url) = &mut module.url {
+                        *url = if self.c8y_endpoint.url_is_in_my_tenant_domain(url.url()) {
+                            // TODO what do we do if this fails?
+                            DownloadInfo::new(
+                                self.auth_proxy.proxy_url(url.url()).unwrap().as_ref(),
+                            )
```

Comment:
What are the possible cause of failures preventing the generation of the appropriate url? This seems to be only a matter of substituting a host by another.

Reply:
I think in practice the only way this will fail is if the URL passed in isn't valid, but if that's the case, we won't consider it a Cumulocity URL and therefore this code will never be called. So I think unwrapping is good here, but it might be worth considering whether this method `proxy_url` wants to take a URL struct rather than parsing the URL internally. That way, we could do the conversion here and it would be clearer why to unwrap (or `url_is_in_my_tenant_domain` could be modified to return `Option<url::Url>` in the case where it is a Cumulocity URL so we entirely avoid parsing it twice).

Reply:
> it might be worth considering whether this method `proxy_url` wants to take a URL struct rather than parsing the URL internally.

This would indeed help error handling and to avoid confusion between "failing to parse the input url" and "failing to substitute the host".